simtools/util/names.py

```diff
@@ -66,27 +178,136 @@ def isValidName(name, allNames):
     return False
 
 
-allTelescopeTypeNames = {
+def validateTelescopeName(name):
+    '''
+    Validate a telescope name.
+
+    Raises
+    ------
+    ValueError
+        If name is not valid.
+
+    Parameters
+    ----------
+    name: str
+
+    Returns
+    -------
+    str
+        Validated name.
+    '''
+    telSite, telClass, telType = splitTelescopeName(name)
+    return telSite + '-' + telClass + '-' + telType
+
+
+def splitTelescopeName(name):
+    '''
+    Split a telescope name into site, class and type.
+
+    Raises
+    ------
+    ValueError
+        If name is not valid.
+
+    Parameters
+    ----------
+    name: str
+        Telescope name.
+
+    Returns
+    -------
+    str, str, str
+        Site (South or North), class (LST, MST, SST ...) and type (any complement).
+    '''
+    nameParts = name.split('-')
+    thisSite = validateSiteName(nameParts[0])
+    telClass = validateName(nameParts[1], allTelescopeClassNames)
+    telType = ''
+    for ii in range(2, len(nameParts)):
+        telType += nameParts[ii] + ('' if ii == len(nameParts) - 1 else '-')
+    return thisSite, telClass, telType
+
+
+def getSiteFromTelescopeName(name):
+    '''
+    Get site name (South or North) from the (validated) telescope name.
+
+    Parameters
+    ----------
+    name: str
+        Telescope name.
+
+    Returns
+    -------
+    str
+        Site name (South or North).
+    '''
+    nameParts = name.split('-')
+    thisSite = validateSiteName(nameParts[0])
+    return thisSite
+
+
+def convertTelescopeNameToYaml(name):
+    '''
+    Get telescope name following the old convention (yaml files) from the current telescope name.
+
+    Parameters
+    ----------
+    name: str
+        Telescope name.
+
+    Returns
+    -------
+    str
+        Telescope name (old convention).
+    '''
+    telSite, telClass, telType = splitTelescopeName(name)
+    newName = telClass + '-' + telType
+    if newName == 'SST-D':
+        return 'SST'
+    elif newName == 'SST-1M':
+        return 'SST-1M'
+    elif newName == 'SST-ASTRI':
+        return 'SST-2M-ASTRI'
+    elif newName == 'SST-GCT':
+        return 'SST-2M-GCT-S'
+    elif newName == 'MST-FlashCam-D':
+        return 'MST-FlashCam'
+    elif newName == 'MST-Nectar-D':
+        return 'MST-NectarCam'
+    elif newName == 'SCT-D':
+        return 'SCT'
+    elif newName in 'LST-D234':
+        return 'LST'
+    elif newName in 'LST-1':
+        return 'LST'
```

Comment:
Hopefully this function won't be necessary very soon anyway, but for now, it might be "prettier" to do this with a dictionary rather than a list of ifs, no?

Reply:
Done, thanks!